src/ui/public/saved_object_table/item_table_state.js

```diff
@@ -0,0 +1,38 @@
+import { SortOrder } from 'ui_framework/components/table/sort_order';
+
+export class ItemTableState {
```

Comment:
I was keeping state and actions separate with the thought of a flux like architecture in the back of my mind.  If you have IDs in here then you end up with multiple IDs arrays  (if I followed the same pattern and split out some of the itemtableactions, then you might have one in an itemsortactions as well). You have to worry about them getting out of sync (@weltenwort- thoughts on this? You mentioned de-syncronization as a concern above).  If you only have one array in state you don't have to worry about that.  I agree passing in itemtablestate is more than this class needs. I could maybe split itemtablestate into itemselectedstate... but if I don't duplicate the ids then it's just an array of selected items?  ... i guess first I want to ask, how do you feel about the separation of state and actions in prep for flux like patterns?   Maybe I could just pass in to all these functions IDs and selected IDs array? Not sure..